File: api/src/controller/dto/party.go
package dto

type PartyCreationRequest struct {
	Code      string `json:"code"`
	Password  string `json:"password"`
	AutoStart bool   `json:"auto-start"`
}

type PartyCreationResponse struct {
	PartyCode string `json:"party-code"`
	AdminCode string `json:"admin-code,omitempty"`
}

type PartyStartRequest struct {
	AdminCode string `json:"admin-code"`
}

File: web/src/api/party.js
import axios from "axios";
import { createPartyURL, connectToPartyURL, startPartyURL } from "../config/env";

export async function createParty(code, password, autoStart) {
    return (await axios.post(createPartyURL(), {
        "code": code,
        "password": password,
        "auto-start": autoStart
    })).data
}

export function connectToParty(code, position, password) {
    const ws = new WebSocket(connectToPartyURL(code, position, password))
    return ws
}

export async function startParty(code, adminCode) {
    return (await axios.post(startPartyURL(code), {
        "admin-code": adminCode
    })).data
}
